### ISSUE TITLE
Soft failure on yml parsing errors

### ISSUE THREAD

**Issue by 4face-studi0** [detekt/detekt#3000]
##  Expected Behavior
1. Being able to build even if some error happens while parsing the yml
2. Have the error logged out to Gradle

## Current Behavior
Build is failing with no recovery

## Context
We have multiple projects and our Core project is providing / hosting the Detekt rule-set.
Problems start when the plugin / rule-set is updated for Core, but for the clients, that would not be a problem unless we're talking about private projects, but they're public projects and users just won't be able to build it without any notice.

Moreover some of them are privately hosted on a private server and and they're just mirrored to GitHub and GitHub may receive updates few weeks after they're actually developed.

**Comment by arturbosch** [comment]
Hi!

Could you elaborate your use case. I'm afraid I did not get it right.
What do you mean with `error happens while parsing the yml`? Yaml syntax errors or deprecated detekt config properties?

**Comment by 4face-studi0** [comment]
Hi @arturbosch 

When I said
> rule-set is updated for Core

I meant updated with new rules added with newer Detekt version.
### So, Detekt x.1 will fail will rule-set for Detekt 0.2

Let me describe it a little more practically ( I'll use fantasy version names ):

1. `Core 0.1` is released: it provides a 
   * Gradle plugin that configure `Detekt 1.0` 
   * A rule-set for `Detekt 1.0` that will be fetch from the Core repo
2. `Client X` start using `Core 0.1 with Detekt 1.0` and fetches `rule-set for Detekt 1.0`
3. `Client X` hits public GitHub repo
4. `Core 0.2` is released, using the newest `Detekt 2.0` and its rule-set ( which includes new rules )
5. User's project will fetch `rule-set for Detekt 2.0` while still using `Detekt 1.0` - its build will fail
 
A solution could be to versionate the rule-set, but it would be pretty annoying: we will have to maintain a lot of files as Detekt is updated pretty frequently and we're trying to keep our rules as much up-to-date as possible.

**Comment by arturbosch** [comment]
hm, sorry I'm still a bit unsure :).
My take is that newer config.yml introduce new rules aka properties in yaml which detekt treats as "unknown" and crashes?
Would:
```yaml
config:
  validation: false
```
help you?

**Comment by 4face-studi0** [comment]
> My take is that newer config.yml introduce new rules aka properties in yaml which detekt treats as "unknown" and crashes?

Yes, that's it! :)

I was not aware of that option! I will try tomorrow and update here!
Thank you

**Comment by 4face-studi0** [comment]
It did solve! :)
Thank you 
I'm closing this.
